pkgs/development/compilers/llvm/18/openmp/gnu-install-dirs.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```

Comment:
i think many of these gnu-install-dirs.patch files can be dropped. there looks like one in the llvm subdirectory which configures rpaths (though don't we use absolute paths anyway) which is different than the rest but the others I don't think are required.

Reply:
It applies so I guess it's still relevant.

Reply:
could you remove the patch and try. i don't think the patch does anything.

Reply:
I'll try without it this evening.

Reply:
So yeah, it builds fine without it afaict. However, this looks sketch:
```
result/lib/libomptarget.so:
	linux-vdso.so.1 (0x0000ffff961b0000)
	libomp.so => /home/ross/ExpidusOS/nixpkgs/result/lib/libomp.so (0x0000ffff95ff0000)
	libLLVM.so.18.1 => /nix/store/vjshafmfx539lqhcpgf87sdpq7i1bdrg-llvm-18.1.1-lib/lib/libLLVM.so.18.1 (0x0000ffff8dfb0000)
	libdl.so.2 => /nix/store/zczdyq0xysvli139lf70i4dydvq6syhk-glibc-2.38-44/lib/libdl.so.2 (0x0000ffff8df80000)
	libstdc++.so.6 => /nix/store/x0nja6gd8pkwv9431dy2c1w5wycipxir-gcc-13.2.0-lib/lib/libstdc++.so.6 (0x0000ffff8dd20000)
	libm.so.6 => /nix/store/zczdyq0xysvli139lf70i4dydvq6syhk-glibc-2.38-44/lib/libm.so.6 (0x0000ffff8dc70000)
	libgcc_s.so.1 => /nix/store/x0nja6gd8pkwv9431dy2c1w5wycipxir-gcc-13.2.0-lib/lib/libgcc_s.so.1 (0x0000ffff8dc30000)
	libc.so.6 => /nix/store/zczdyq0xysvli139lf70i4dydvq6syhk-glibc-2.38-44/lib/libc.so.6 (0x0000ffff8da70000)
	/nix/store/cv8mfy5wdfwfw4iwhdlkl4ddy8apl667-glibc-2.38-27/lib/ld-linux-aarch64.so.1 (0x0000ffff96178000)
	libffi.so.8 => /nix/store/ipvswqwv0rslqhni4gk0s11w42kzw7n1-libffi-3.4.6/lib/libffi.so.8 (0x0000ffff8da30000)
	librt.so.1 => /nix/store/zczdyq0xysvli139lf70i4dydvq6syhk-glibc-2.38-44/lib/librt.so.1 (0x0000ffff8da00000)
	libz.so.1 => /nix/store/1ch4q190znxnxdbqy278f73fs7834b5g-zlib-1.3.1/lib/libz.so.1 (0x0000ffff8d9c0000)
	libncursesw.so.6 => /nix/store/0d51qvm73z3gw6sdd9anzgr0l2ffaajr-ncurses-6.4/lib/libncursesw.so.6 (0x0000ffff8d930000)
	libxml2.so.2 => /nix/store/8bj9m4jhq1193db8xxmfax5jnsg1i5f8-libxml2-2.12.5/lib/libxml2.so.2 (0x0000ffff8d7c0000)
```
libomp.so should be referencing the nix path and not the result symlink imo.

Reply:
that doesn't seem like a result of removing the patch. looking at llvm17 it has the same issue.
```
result/lib/libomptarget.so.17:
        linux-vdso.so.1 (0x00007ffd5ade4000)
        libomp.so => /home/anna/src/nix.4/result/lib/libomp.so (0x00007c5e54dc9000)
        libLLVM-17.so => /nix/store/a57hwpz8g9m990sbmlc6ms9v3sjhpn3k-llvm-17.0.6-lib/lib/libLLVM-17.so (0x00007c5e4c200000)
        libdl.so.2 => /nix/store/1rm6sr6ixxzipv5358x0cmaw8rs84g2j-glibc-2.38-44/lib/libdl.so.2 (0x00007c5e54dc4000)
        libstdc++.so.6 => /nix/store/agp6lqznayysqvqkx4k1ggr8n1rsyi8c-gcc-13.2.0-lib/lib/libstdc++.so.6 (0x00007c5e4be00000)
        libm.so.6 => /nix/store/1rm6sr6ixxzipv5358x0cmaw8rs84g2j-glibc-2.38-44/lib/libm.so.6 (0x00007c5e54ce0000)
        libgcc_s.so.1 => /nix/store/agp6lqznayysqvqkx4k1ggr8n1rsyi8c-gcc-13.2.0-lib/lib/libgcc_s.so.1 (0x00007c5e54cbb000)
        libc.so.6 => /nix/store/1rm6sr6ixxzipv5358x0cmaw8rs84g2j-glibc-2.38-44/lib/libc.so.6 (0x00007c5e4bc17000)
        /lib64/ld-linux-x86-64.so.2 (0x00007c5e54f0f000)
        libffi.so.8 => /nix/store/wb7d5iw5f08qbm6p72pz1mdd4qgcb7gs-libffi-3.4.6/lib/libffi.so.8 (0x00007c5e54caa000)
        librt.so.1 => /nix/store/1rm6sr6ixxzipv5358x0cmaw8rs84g2j-glibc-2.38-44/lib/librt.so.1 (0x00007c5e54ca3000)
        libz.so.1 => /nix/store/zbaajn0dk65kwvrcglhli1j4xx4vhvvj-zlib-1.3.1/lib/libz.so.1 (0x00007c5e54c85000)
        libncursesw.so.6 => /nix/store/1n8dg5f4435qm73m4b9m0479igx7iq7l-ncurses-6.4/lib/libncursesw.so.6 (0x00007c5e54c11000)
        libxml2.so.2 => /nix/store/6wfv81r4danww91b2p3i7i46l7n1lcs1-libxml2-2.12.5/lib/libxml2.so.2 (0x00007c5e4c0a0000)
``

Reply:
it is just using `$ORIGIN` in the rpath.
```
$ patchelf --print-rpath libomptarget.so.17
$ORIGIN:/nix/store/fq1h21fddgn18aydi85knvwg7xnjc3iz-openmp-17.0.6/lib:/nix/store/a57hwpz8g9m990sbmlc6ms9v3sjhpn3k-llvm-17.0.6-lib/lib:/nix/store/1rm6sr6ixxzipv5358x0cmaw8rs84g2j-glibc-2.38-44/lib:/nix/store/agp6lqznayysqvqkx4k1ggr8n1rsyi8c-gcc-13.2.0-lib/lib
```

Reply:
Oh, interesting. Well, it built fine without the patch so it must be okay to remove then.

Reply:
looking at the others:
```console
$ fd gnu pkgs/development/compilers/llvm/18
pkgs/development/compilers/llvm/18/clang/gnu-install-dirs.patch
pkgs/development/compilers/llvm/18/compiler-rt/gnu-install-dirs.patch
pkgs/development/compilers/llvm/18/lld/gnu-install-dirs.patch
pkgs/development/compilers/llvm/18/llvm/gnu-install-dirs-polly.patch
pkgs/development/compilers/llvm/18/llvm/gnu-install-dirs.patch
pkgs/development/compilers/llvm/18/openmp/gnu-install-dirs.patch
```
it looks like all can be removed except for `llvm/gnu-install-dirs.patch` which seems to do much more than the other patches.

Reply:
Clang's looks like it shouldn't be removed too.

Reply:
it might be fine to remove. it looks the the base of this patch came in at:
`pkgs/development/compilers/llvm/15/clang/gnu-install-dirs.patch`

and i'd guess that the code around it got removed but the patch was modified to add it back and it probably is not required.

Reply:
Clang's definitely shouldn't be removed, just tried it and got a bad build. The install phase didn't work right and the mv call failed.

Reply:
<details><summary>so it looks like that whole patch is to just avoid doing `mv $out/lib $out/lib` in post install</summary>

```diff
diff --git a/pkgs/development/compilers/llvm/18/clang/default.nix b/pkgs/development/compilers/llvm/18/clang/default.nix
index 01b33c540..8da3d0e31 100644
--- a/pkgs/development/compilers/llvm/18/clang/default.nix
+++ b/pkgs/development/compilers/llvm/18/clang/default.nix
@@ -50,8 +50,6 @@ let
 
     patches = [
       ./purity.patch
-      # https://reviews.llvm.org/D51899
-      ./gnu-install-dirs.patch
       ../../common/clang/add-nostdlibinc-flag.patch
       (substituteAll {
         src = ../../common/clang/clang-at-least-16-LLVMgold-path.patch;
@@ -70,8 +68,7 @@ let
     postInstall = ''
       ln -sv $out/bin/clang $out/bin/cpp
 
-      mkdir -p $lib/lib/clang
-      mv $lib/lib/${lib.versions.major version} $lib/lib/clang/${lib.versions.major version}
+      mv $out/lib $lib/lib
 
       # Move libclang to 'lib' output
       moveToOutput "lib/libclang.*" "$lib"
```
</details>

if the patch is desirable for some reason, for example if library location gets used at compile time (have to check but don't think it does) it should use the correct directory in `set(header_install_dir ${CMAKE_INSTALL_LIBDIR}${LLVM_LIBDIR_SUFFIX}/${CLANG_VERSION_MAJOR}/include)` -> `set(header_install_dir ${CMAKE_INSTALL_LIBDIR}${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION_MAJOR}/include)`  so the `mkdir -p $lib/clang` and following `mv` can be deleted.

Reply:
but it looks like the generated cmake files use the patches so they should stay.